cdci_data_analysis/analysis/instrument.py

```diff
@@ -716,6 +731,33 @@ def set_input_products(self, par_dic, input_file_path,input_prod_list_name):
 
         return has_prods
 
+    def upload_files_request(self, par_dic, request, upload_dir, products_url, token=None):
+        list_uploaded_files = []
+        if request.method == 'POST':
+            if validators.url(products_url):
+                basepath = os.path.join(products_url, 'dispatch-data/download_file')
+            else:
+                basepath = os.path.join(products_url, 'download_file')
+            for f in request.files:
+                # TODO needed since those two files are extracted in a previous step
+                if f != 'user_scw_list_file' and f != 'user_catalog_file':
+                    f_path = upload_file(f, upload_dir)
+                    if f_path is not None:
+                        file_hash = make_hash_file(f_path)
+                        f_name, f_ext = os.path.splitext(os.path.basename(f_path))
+                        new_file_name = f_name.split('.')[0] + '_' + file_hash + f_ext
+                        new_file_path = os.path.join(upload_dir, new_file_name)
+                        list_uploaded_files.append(new_file_name)
+                        os.rename(f_path, new_file_path)
+                        dpars = urlencode(dict(file_list=new_file_name,
+                                               query_status="ready",
+                                               return_archive=False,
+                                               token=token))
```

Comment:
True, good point. Perhaps it can be added within the service nb2workflow ? I'd avoid it in the plugin...

Reply:
In nb2workflow we don't currently have it. So appending it somewhere in the dispatcher could make sense. 

On the other hand, it will also affect the job identifier on the nb2workflow side

Reply:
> In nb2workflow we don't currently have it. So appending it somewhere in the dispatcher could make sense.

like after the job_id generation ? an option we have already discussed

> 
> On the other hand, it will also affect the job identifier on the nb2workflow side

in which way?

Reply:
The url is now built without the token, rather with a placeholder, this will be substituted within the plugin (https://github.com/oda-hub/dispatcher-plugin-nb2workflow/pull/94) if needed.

This is for not impacting the `job_id` generation

Reply:
> in which way?

Imagine the token is changed (e.g. regenerated with the new lifetime). 
On the dispatcher side, the job_id is still the same, because the placeholder is used when calculating it.
But then the placeholder is substituted (in the plugin before requesting the backend), so nb2workflow backend will see this request as a new one (the job_id there is calculated separately) and will start a new job.

Reply:
ok I wasn't aware of this re-calculation of the job_id
so you suggest to leave the placeholder here, and not making any token substitution on the plugin? then the nb2workflow backend will anyway have access to the token, and can itself make the substitution ?

Reply:
btw where is the job_id calculation happening in the backend?

Reply:
> so you suggest to leave the placeholder here, and not making any token substitution on the plugin? then the nb2workflow backend will anyway have access to the token, and can itself make the substitution ?

I don't fully like it because we try to keep the backend nb2workflow standalone. Like this we bound some of its code to dispatcher. But I don't see any general approach

 
> btw where is the job_id calculation happening in the backend?

It's here https://github.com/oda-hub/nb2workflow/blob/d68c869f138087936cdb02125ba486a46f3834b1/nb2workflow/service.py#L285. It has slightly different implementation but the idea is the same

Reply:
ok, and can the token be removed before the generation of the key ? or is it meant to stay there...? we remove on the dispatcher (but they are two separate entities)

https://github.com/oda-hub/dispatcher-app/blob/8d519bf453c42443e22cb5dadcbf24a71e49d706/cdci_data_analysis/flask_app/dispatcher_query.py#L676-L702

Reply:
> ok, and can the token be removed before the generation of the key ?

It can, in principle. The one that proposed to pass the current token is an underscored argument `_token`. It's removed by default. This one is a part of other argument without fixed name. So the logic can be that we remove "token" arg from any argument that has a form of url before calculating id. Not sure, this won't produce any side effects

Reply:
as discussed, the token will be injected inside on the nb2workflow service, after the parsing of the url